### DISCUSSION TITLE
About RSA key generation.

### DISCUSSION THREAD

**Discussion by Liu-ErMeng** [openssl/openssl#24713] [Q&A] [answered]
Why openssl keep the `rsa_multiprime_keygen` if the `key length <2048 or  public exponent <= 2^16` ？It's a performance consideration (It's faster to use `rsa_multiprime_keygen`  when the key length lower than 2048), or it's just keep for the multi-prime purpose?

https://github.com/openssl/openssl/blob/master/crypto/rsa/rsa_gen.c#L615

**Comment by t8m** [comment, marked as answer]
The ossl_rsa_sp800_56b_generate_key() does not support these parameters, so that's why. The rsa_multiprime_keygen() also by default uses the standard 2 primes.

**Reply by jchx-c** [reply]
Does the implementation of the rsa_multiprime_keygen() follow a standard?

**Reply by t8m** [reply]
In what sense? It generates correct 2 or more prime RSA keys so yeah it is following the RSA standard in this sense.

It does not follow NIST standards for RSA though.
